### PR TITLE
Add symbols for event mons and refactor seteventmon

### PR DESCRIPTION
**Pull request by luctowers**
There is currently a problem with this PR.

Latios and Latias use two different `seteventmon` commands, but they share the same flag. I've just appended `_LATIOS` and `_LATIAS` for now (which fails extraction because those flags don't exist). What do you think the best solution is? Create two more flags, one for each and set them accordingly?

I tested this on multiple event encounters.

### CODE REVIEW

**Comment by Zunawe** [comment]
Hmm. Perhaps it's best to leave the Latis alone for the moment. We could in the future remove the choice that Emerald gives you for which one is the roamer and which one is at the island. And then add both the roamer and island encounters to this list then. Or maybe reference some field in `gArchipelagoOptions` for every instance or something.

Or, for now, if you want to use a dummy flag for each of them just to get them on the static encounter list that could be fine. Keep the existing actual flags there but change the target name to a couple of explicitly unused flags. And in the future when we're ready to handle them correctly we can change the target names back to whatever they need to be.

That way they'll exist in the "shuffle" pool for static encounters. And then later down the line when it's possible to get to the island during a rando or when legendary hunt is a goal, we'll actually decide how to handle them.

File: src/battle_setup.c [hunk 1]
```diff
@@ -543,6 +557,8 @@ void BattleSetup_StartLegendaryBattle(void)
     case SPECIES_MEW:
         CreateBattleStartTask(B_TRANSITION_GRID_SQUARES, MUS_VS_MEW);
         break;
+    default:
+        CreateBattleStartTask(lengendary_transitions[Random2()%5], lengendary_music[Random2()%5]);
```
Comment (Zunawe) [inline src/battle_setup.c:561]: "lengendary"

I might tweak this line in the future to satisfy my own ideas, but I'm happy to merge it and leave it as-is at the moment.

Comment (luctowers) [reply src/battle_setup.c]: > Or, for now, if you want to use a dummy flag for each of them just to get them on the static encounter list that could be fine. Keep the existing actual flags there but change the target name to a couple of explicitly unused flags. And in the future when we're ready to handle them correctly we can change the target names back to whatever they need to be.
 
Going this route for now.

**Review by Zunawe — APPROVED** [review]
Looks good. Thank you

**Pull request merged**